### DISCUSSION TITLE
Support of the on-premise environments

### DISCUSSION THREAD

**Discussion by filonenkoalex** [Power-Maverick/DataverseDevTools-VSCode#167] [Q&A]
Hey guys! Awesome work! 

Can we support on-premise environments? 

On-premise environments (both 9.0 and 9.1) with enabled OAuth should work as an extension already support same concept.

```http request for on-prem 
POST https://<ADFS>/adfs/oauth2/token
Content-Type: application/x-www-form-urlencoded
 
grant_type=password&username=<username>&password=<password>&scope=openid <CrmUrl>//user_impersonation
 
HTTP/1.1 200 OK
 
{"access_token":"eyJ0***","token_type":"bearer","expires_in":28800,"resource":"<CrmUrl>/","refresh_token":"Qcq-****","refresh_token_expires_in":83999,"scope":"openid","id_token":"eyJ0****"}
```

I understand the complexity of trying to support legacy, but it could be helpful to many developers who are stuck with on-prem.
Let me know if there is any other option to connect to on-prem or if any internal decisions are made about this.

**Comment by Power-Maverick** [comment]
On-prem connections has lots of complexities plus I don't have an on-prem environment to teat and based on the user base other items is getting more priority. If you want please feel free to submit a PR for on-prem connections.
